external/implementations/samlconf-openam-impl/src/main/java/org/codice/compliance/saml/plugin/openam/OpenAMDataParser.java

```diff
@@ -0,0 +1,124 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+class OpenAMDataParser {
+
+  //  A class only for retrieving values from the openAM data file.
+  private final ImmutableMap<String, String> formMapping;
+  private final String openAMRealm;
+  private final String baseURL;
+
+  private static final String REALM = "realmName";
+  private static final String BASE_URL = "baseUrl";
+  private static final String LOGIN_FORM = "loginForm";
+  private static final String INPUT = "input-";
+  private static final String OUTPUT = "output-";
+
+  private static final String USER_ARG = "USER.ARG";
+  private static final String PASSWORD_ARG = "PASS.ARG";
+
+  private static final String fileName = "OpenAMLoginData.xml";
+
+  OpenAMDataParser() {
+
+    DocumentBuilder doc;
+    try {
+      DocumentBuilderFactory docBF = DocumentBuilderFactory.newInstance();
+      doc = docBF.newDocumentBuilder();
+    } catch (ParserConfigurationException e) {
+      throw new IllegalStateException(e);
+    }
+
+    Document xml;
+    try (InputStream xmlFile = getClass().getClassLoader().getResourceAsStream(fileName)) {
+      if (xmlFile == null) {
+        throw new IllegalStateException(fileName + " was not found");
+      }
+      xml = doc.parse(xmlFile);
+
+    } catch (IOException | SAXException e) {
+      throw new IllegalStateException(e);
+    }
+
+    //  Parsing the XML login data
+    String tempRealm = xml.getElementsByTagName(REALM).item(0).getTextContent();
+    openAMRealm = (tempRealm.equals("/")) ? "" : tempRealm;
+    baseURL = xml.getElementsByTagName(BASE_URL).item(0).getTextContent();
+
+    //  Since the authentication payload sent is customizable by the client of OpenAM
+    //  This parsing has to be dynamic. Loop through the login form tag and continue while
+    //  both the Input and Output tags are supplied.
+    Element loginInfo = (Element) xml.getElementsByTagName(LOGIN_FORM).item(0);
+
+    Map<String, String> tempMap = new HashMap<>();
+    int i = 1;
+    while (hasChildTag(loginInfo, INPUT + i) && hasChildTag(loginInfo, OUTPUT + i)) {
+
+      tempMap.put(
+          loginInfo.getElementsByTagName(INPUT + i).item(0).getTextContent(),
+          expandArg(loginInfo.getElementsByTagName(OUTPUT + i).item(0).getTextContent()));
+      i++;
+    }
+
+    formMapping = ImmutableMap.copyOf(tempMap);
+  }
+
+  private boolean hasChildTag(Element element, String tagName) {
+    return element.getElementsByTagName(tagName).getLength() != 0;
+  }
+
+  private String expandArg(String arg) {
+    //  If they didn't supply the system property just return.
+    String userLogin = System.getProperty(USER_LOGIN);
```

Comment:
✏️ Should be able to get this once, on instance construction, and save the repeated calls to read the system property.

Reply:
You can use `username` in `TestCommon` instead.

Reply:
Sounds like a plan

Reply:
Because the username and password is required by OpenAM I'm going to let it this throw an error if the system property is not set.

Reply:
Maybe you should have a default for OpenAM like all the other IdPs that you document. I know admin/admin was too short so something just to be consistent.